src/game_events/conditional_wml.hpp

```diff
@@ -24,11 +24,12 @@
 class vconfig;
 
 #include "config.hpp"
+#include "units/unit.hpp"
 
 namespace game_events
 {
 	bool conditional_passed(const vconfig& cond);
-	bool matches_special_filter(optional_const_config cfg, const vconfig& filter);
+	bool matches_special_filter(const_attack_ptr attack, const vconfig& filter);
```

Comment:
I think this function could be removed, with the lines in handlers.cpp changed to directly `return swf_.empty() || attack->matches_filter(swf_.get_parsed_config());`

From the implementation of the function:
* The `if (!attack)` block is no longer necessary, as we know that `std::make_shared` doesn't return `nullptr`.
* The comment about putting `auto attack` on the heap might still be useful, but belongs on the line before `attack = std::make_shared`, because that's what that comment refers to.
* After removing those, there's only one line left in `matches_special_filter`.